draft-ietf-anima-brski-prm.md

```diff
@@ -928,99 +945,106 @@ The fields contained in the `pledgevoucherrequesttrigger` are:
-
-The fields contained in the `prmasd` are:
+{: #prmasd_CDDL_def title='CDDL for BRSKI-PRM Agent Signed Data' artwork-align="left"}
 
-* `created-on`: creation date and time as standard date/time string as defined in {{!RFC3339}} 
+The `created-on` member SHALL contain the creation date and time as standard date/time string as defined in {{Section 5.6 of !RFC3339}}.
 
-* `serial-number`: product-serial-number in the X520SerialNumber field of the IDevID certificate of the pledge as string as defined in {{Section 2.3.1 of !RFC8995}}
+The `serial-number` member SHALL contain the product-serial-number of the pledge with which the Registrar-Agent assumes to communicate as string.
+The format MUST correspond to the X520SerialNumber field of IDevID certificates.
 
-{{prmasd_payload}} below shows an example for unsigned BRSKI-PRM Agent Signed Data in JSON syntax. 
+{{prmasd_payload}} below shows an example for the BRSKI-PRM Agent-Signed Data:
 
 ~~~~
 {
   "created-on": "2021-04-16T00:00:01.000Z",
   "serial-number": "callee4711"
 }
 ~~~~
-{: #prmasd_payload title="Data example for prmasd" artwork-align="left"}
+{: #prmasd_payload title="BRSKI-PRM Agent-Signed Data Example" artwork-align="left"}
+
+#### JWS Protected Header
 
-The JWS Protected Header of the `agent-signed-data` JWS structure MUST contain the following parameters (see {{asd_header}} for an example):
+The JWS Protected Header of the `agent-signed-data` member MUST contain the following standard Header Parameters as defined in {{RFC7515}}:
 
-* `alg`: algorithm type used to create the signature, e.g., `ES256` as defined in {{Section 4.1.1 of RFC7515}}
+* `alg`: SHALL contain the algorithm type used to create the signature, e.g., `ES256` as defined in {{Section 4.1.1 of RFC7515}}
+* `kid`: SHALL contain the base64-encoded bytes of the SubjectKeyIdentifier (the `KeyIdentifier` OCTET STRING value) of the Registrar-Agent EE certificate as defined in {{Section 4.2.1.2 of !RFC5280}}
 
-* `kid`: base64-encoded bytes of the SubjectKeyIdentifier (the "KeyIdentifier" OCTET STRING value) of the EE (RegAgt) certificate.
+{{asd_header}} below shows an example for this JWS Protected Header:
 
 ~~~~
 {
   "alg": "ES256",
   "kid": "base64encodedvalue=="
 }
 ~~~~
-{: #asd_header title="Protected Header example inside agent-signed-data" artwork-align="left"}
-
-Note that at the time of receiving the PVR trigger, the pledge cannot verify the registrar LDevID certificate and has no proof-of-possession of the corresponding private key for the certificate.
-Hence, the tPVR is an unsigned artifact and the pledge only accepts the registrar LDevID certificate provisionally until it receives the voucher as described in {{voucher}}.
-
-The pledge will also be unable to verify the agent-signed-data itself as it does not possess the EE (RegAgt) certificate and the domain trust has not been established at this point of the communication.
-Verification SHOULD be done, after the voucher has been received.
-
```

Comment:
I would propose to keep the above as background information, why the agent-signed-data is only passed through the pledge and not processed.

Reply:
We already explained this for the general artifact definition in line 919 -> 921 ("accept the parameters provisionally", see comment above).
I rather want the normative statement at the correct point, where the voucher is being received by the pledge, so in Section 7.6 (still need to add it there once I reach that section).

Reply:
Okay, understood. Lines 919-921 are also fine. I like the structured approach which treats every exchange alike in the describing structure.